source/common/http/codec_client.cc

```diff
@@ -23,27 +25,40 @@ CodecClient::CodecClient(Type type, Network::ClientConnectionPtr&& connection,
   ENVOY_CONN_LOG(debug, "connecting", *connection_);
   connection_->connect();
 
+  if (idle_timeout_) {
+    idle_timer_ = dispatcher.createTimer([this]() -> void { onIdleTimeout(); });
+    enableIdleTimer();
+  }
+
   // We just universally set no delay on connections. Theoretically we might at some point want
   // to make this configurable.
   connection_->noDelay(true);
 }
 
 CodecClient::~CodecClient() {}
 
-void CodecClient::close() { connection_->close(Network::ConnectionCloseType::NoFlush); }
+void CodecClient::close() {
+  connection_->close(Network::ConnectionCloseType::NoFlush);
+  disableIdleTimer();
```

Comment:
I don't understand why this was added back here. What you had before was correct. Close should absolutely raise events, unless the connection was already closed somehow. There is likely some bug in the test code if only the mock test was failing. Can you please investigate?

Reply:
Ok. The reason the test code using a MockConnection which is not raising the callbacks. The connnection callbacks need to be raised specifically and rest of the tests do the same. Raised them and it works. PTAL.